cli/cmdlineparser.cpp

```diff
@@ -290,10 +290,14 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
             }
 
             else if (std::strncmp(argv[i], "--cppcheck-build-dir=", 21) == 0) {
-                // TODO: bail out when the folder does not exist? will silently do nothing
                 mSettings.buildDir = Path::fromNativeSeparators(argv[i] + 21);
                 if (endsWith(mSettings.buildDir, '/'))
                     mSettings.buildDir.pop_back();
+
+                if (!Path::directoryExists(mSettings.buildDir)) {
```

Comment:
small nit: we should not assign anything to the settings unless we know it is valid. We don't do that consistently and it should not matter at all since we bail out afterwards anyways but it would be cleaner. No need to change anything here.